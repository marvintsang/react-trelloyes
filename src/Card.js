import React from 'react'
import './Card.css'

class Card extends React.Component{
    render(){
        //Individual card information
        let currentCard = this.props.currentCard;
        //console.log(currentCard);
        //console.log(this.props.allCards[currentCard])
        return(
            <div key={this.props.key} 
                id={this.props.currentCard}
                className='Card'
            >
                <button type='button'>Delete</button>
                <h3>{this.props.allCards[currentCard].title}</h3>
                <p>{this.props.allCards[currentCard].content}</p>
                
            </div>
        )
    }
}


export default Card