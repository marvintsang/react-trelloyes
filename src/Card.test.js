import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card'
import renderer from 'react-test-renderer'

describe('smoke test', () => {
    it('card test', () => {
        //Are the cards displaying the way that they should be
        //Expecting props, can be any values
      const div = document.createElement('div');
      ReactDOM.render(<Card />, div);
      ReactDOM.unmountComponentAtNode(div);
      });

    it('snap test', ()=>{
        //making sure it loads with the components it needs
    const obj = {one:{title:'thetitle', content:'thecontent'}}
    const tree = renderer.create(<Card currentCard='one' allCards={obj} />).toJSON();
    expect(tree).toMatchSnapshot()

    })
})
