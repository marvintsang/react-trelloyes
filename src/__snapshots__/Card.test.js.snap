// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke test snap test 1`] = `
<div
  className="Card"
  id="one"
>
  <button
    type="button"
  >
    Delete
  </button>
  <h3>
    thetitle
  </h3>
  <p>
    thecontent
  </p>
</div>
`;
